inspirehep/modules/multieditor/actions.py

```diff
@@ -0,0 +1,299 @@
+                    record[key] = [self.value if x == self.value_to_check else x for x in record[key]]
+
+                elif self.match_type == 'contains':
+                    record[key] = [self.value if self.value_to_check in x else x for x in record[key]]
+                self.changed = True
+            else:
+                if self.match_type == 'is equal to' and\
+                        record[key] == self.value_to_check:
+                    record[key] = self.value
+                if self.match_type == 'matches regular expression' and\
+                        re.search(
+                            re.escape(self.value_to_check),
+                            record[key]):
+                                record[key] = self.value
+                elif self.match_type == 'contains' and\
+                        self.value_to_check in record[key]:
+                    record[key] = self.value
+                self.changed = True
+                return
+        else:
+            if isinstance(record[key], list):
+                for array_record in record[key]:
+                    self.apply_action(array_record, new_schema, position + 1, checked)
+            else:
+                self.apply_action(record[key], new_schema, position + 1, checked)
+
+
+def create_schema_record(schema, path, value):
+    """Object creation in par with the schema."""
+    record = {}
+    temp_record = record
+    new_schema = schema
+    if new_schema['type'] == 'array':
+        new_schema = new_schema['items']['properties']
+    elif new_schema['type'] == 'object':
+        new_schema = new_schema['properties']
+    for key in path:
+        new_schema = new_schema[key]
+        if new_schema['type'] == 'object':
+            new_schema = new_schema['properties']
+            temp_record[key] = {}
+            temp_record = temp_record[key]
+
+        elif new_schema['type'] == 'array':
+            if new_schema['items']['type'] == 'object':
+                new_schema = new_schema['items']['properties']
+            if len(path) == 1:
+                temp_record[key] = [value]
+                return record
+            else:
+                temp_record[key] = [{}]
+                temp_record = temp_record[key][0]
+    temp_record[path[-1]] = value
+    return record
+
+
+def check_value(record, match_type, keys, value_to_check, position):
+    """Where continues to find the value."""
+    key = keys[position]
+    if not record.get(key):
+        if match_type == 'does not exist':
+            return True
+        else:
+            return False
+    temp_record = record[key]
+    if isinstance(temp_record, list):
+        for index, array_record in enumerate(temp_record):
+            if position + 1 == len(keys):
+                if match_type == 'is equal to' and\
+                        array_record == value_to_check:
+                    return True
+                elif match_type == 'contains' and\
```

Comment:
Even though it might go above the line limit, in this case it would be more readable to not break the line.